module-4/backend/week_three.md

```diff
@@ -1,19 +1,73 @@
 ## Week Three - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. At a high level, what is Node?
+
+>Node is a JavaScript runtime built on Chrome's V8 engine.
+
 2. What is Express? What is Express similar to in the Ruby world?
+
+>Express is a lightweight framework to help organize your web application into an MVC architecture on the server side.
+
 3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+
+>`app.get('/api/v1/meals/:meal_id/foods/:food_id', data => {
+    $('input[type=checkbox]').prop('checked',false);
+    $(`table#${mealId} tr:not(:first-child)`).remove()
+    mealFoods(data)
+    }).then(() => {
+        $('tr.all-total').remove()
+        allTotalCal()
+      })
+      .then(() => {
+        $('tr.all-rem').remove()
+        allRemCal()
+    })
+
+})`
+
 4. What do we use Knex for?
+
+>Framework to make queries to the databaase and handle the promises from those queries. (`raw`, `.transaction`, `.insert`). It acts
+alot like ActiveRecord in that it assissts you in manipulating the database.
+
 5. How **could** you organize your code to follow the MVC design pattern in your Quantified Self project?
+
+>For QS we did it as model and controller. The controller directory handled the routes and the model acted like a Ruby model class,
+Instantiating the meal and food classes.
+
 6. How do you execute raw SQL in node?
+
+> `database.raw('SQL query')`
+
 7. What are some advantages to having your front end and back end code live in separate applications? What are some disadvantages?
 
-#### Review  
+>This way they are seperate and if one breaks it won't affect the other one. The downside I think would be that it can get confusing to a
```

Comment:
You can have separate backend and front ends with other languages besides javascript such as a Rails backend for instance.